### PR TITLE
Port a few fixures from node-tar

### PR DESCRIPTION
**Pull request by am11**
Copy a few fixtures from https://github.com/npm/node-tar/tree/main/test/fixtures/tars which `TarReader` was failing to read.

Ref: https://github.com/dotnet/runtime/issues/74316

cc @danmoseley, @carlossanlop

### CODE REVIEW

**Comment by danmoseley** [comment]
What do you think of including them all? There's only 16 / 92K. We can ensure we continue to successfully open the others.

File: THIRD-PARTY-NOTICES.TXT [hunk 1]
```diff
@@ -87,4 +87,23 @@ License for ibm-fpgen (https://github.com/nigeltao/parse-number-fxx-test-data)
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
-   limitations under the License.
\ No newline at end of file
+   limitations under the License.
+
+License for node-tar (https://github.com/npm/node-tar)
```
Comment (danmoseley) [inline THIRD-PARTY-NOTICES.TXT:92]:
```suggestion
License for node-tar (https://github.com/npm/node-tar) test files

```

**Comment by danmoseley** [comment]
@am11 I added all the golang ones as well https://github.com/dotnet/runtime-assets/pull/264

should merge clean.

**Comment by danmoseley** [comment]
I went ahead and pushed up the others. for some reason my copy of next-file-has-long.tar differs from the one you had in the PR? LMK if we should use yours for some reason. Surely not a line endings thing.

**Review by carlossanlop — APPROVED** [review]
LGTM. Thank you @am11 and @danmoseley !

**Pull request merged**